lib/experiments/index.js

```diff
@@ -1,5 +1,16 @@
 'use strict';
 
+const path = require('path');
+const findup = require('find-up');
+const Yam = require('yam');
+
+const cliPath = findup.sync('.ember-cli');
+const emberCli = new Yam('ember-cli', {
+  primary: path.dirname(cliPath),
+});
+const experimentsInDotCli = emberCli.get('experiments') || {};
+Object.assign(process.env, experimentsInDotCli);
```

Comment:
Hmm, I'd prefer to do this where we do the rest of the yam stuff ([around here](https://github.com/ember-cli/ember-cli/blob/0ef3f1b9ce8450ccb87dc61cd87547a4ce280ad7/lib/cli/index.js#L78)). I also think that if we are "just" merging with `process.env` we shouldn't use `experiments` as the key in `.ember-cli` (we would use `environment` or something else)...

Reply:
That was my original intention as well but there was an issue with that.

We require models/project [here](https://github.com/astronomersiva/ember-cli/blob/master/lib/cli/index.js#L77) which requires lib/experiments internally. Due to this, the experiments are set even before the `.ember-cli` is loaded.

Placing the Project require below the getConfig also won't help as getConfig again uses lib/experiments internally.

I agree that the key name has to be changed to `environment` in `.ember-cli`.